metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -1386,17 +1386,28 @@ def _build_simple_metric_recipe(
                 0
             ].time_granularity
 
-            # If the smallest queried grain is equal to the offset grain, join after aggregation. Otherwise, the grain
-            # needed for offset will not be available anymore, so join before aggregation.
+            # If the smallest queried grain is less than or equal to the offset grain, join after aggregation.
+            # Otherwise, the grain needed for offset will not be available anymore, so join before aggregation.
             join_to_time_spine_description = JoinToTimeSpineDescription(
                 join_type=SqlJoinType.INNER,
                 offset_window=child_metric_offset_window,
                 offset_to_grain=child_metric_offset_to_grain,
             )
             if (
                 offset_grain
-                and smallest_queried_agg_time_grain == offset_grain
-                and not offset_grain.is_custom_granularity  # custom offset window has special logic handled later
+                and smallest_queried_agg_time_grain
+                # We can't know if the custom grain, once aggregated, will be larger or smaller than the offset grain.
+                and not smallest_queried_agg_time_grain.is_custom_granularity
+                # Custom offset windows have special logic handled later.
+                and not offset_grain.is_custom_granularity
+                # If queried grain > offset grain, offset grain won't be available after aggregation.
+                and smallest_queried_agg_time_grain.base_granularity.to_int() <= offset_grain.base_granularity.to_int()
+                # Special case: WEEK does not roll up to the larger granularities cleanly, so we can't apply WEEK
+                # aggregation before offset_to_grain join.
+                and not (
+                    smallest_queried_agg_time_grain.base_granularity is TimeGranularity.WEEK
+                    and child_metric_offset_to_grain
+                )
```

Comment:
Note about tests for this scenario: I did not add any new tests for it, but there were some check query tests that already existed for this case. They failed without this special handling, showing completely null results. Those tests now passing due to this special handling indicate that this is working how we want.